social_core/backends/mediawiki.py

```diff
@@ -0,0 +1,175 @@
+"""
+MediaWiki OAuth1 backend
+========================
+
+Usage
+-----
+
+In addition to the general setup you need to define the
+following parameters. In Django's settings.py you would
+use for English Wikipedia::
+
+    SOCIAL_AUTH_MEDIAWIKI_KEY = <consumer_key>
+    SOCIAL_AUTH_MEDIAWIKI_SECRET = <consumer_secret>
+    REQUEST_TOKEN_URL =
+        'https://en.wikipedia.org/w/index.php?title=Special:OAuth/initiate'
+    AUTHORIZATION_URL =
+        'https://en.wikipedia.org/w/index.php?title=Special:Oauth/authorize'
+    ACCESS_TOKEN_URL =
+        'https://en.wikipedia.org/w/index.php?title=Special:Oauth/token'
+    IDENTIFY_URL =
+        'https://en.wikipedia.org/w/index.php'
+
+If users should only be prompeted once for permission use:
```

Comment:
The mwoauth library took a slightly different approach and chose to always use the `authenticate` endpoint (See https://github.com/mediawiki-utilities/python-mwoauth/commit/52285284254179d2990bb59cac5ef2605b6b9ced). One benefit of this is that much of the work you are doing later to extract the title and merge it into other parameters is eliminated. Instead that library only needs a base URI like `https://en.wikipedia.org/w/index.php` and it can derive all of the URLs needed from that.